libs/pulumi/src/executors/up/executor.ts

```diff
@@ -69,31 +67,58 @@ export default async function runUpExecutor(
         }
     }
 
-    const pulumiArguments = process.argv.slice(4)
+    const commandLineMap: Record<string, string> = {
+        '--disableIntegrityChecking': '--disable-integrity-checking',
+        '--nonInteractive': '--non-interactive',
+        '--targetDependents': '--target-dependents',
+        '--targetReplace': '--target-replace',
+    }
+
+    let env: string | undefined
+
+    // NX Mangles command line args. Let's fix them back.
+    // https://github.com/nrwl/nx/issues/5710
+    // This is an incomplete list, should do all args
+    const pulumiArguments = process.argv
+        .slice(4)
+        .map((arg) => {
+            const mangled = Object.keys(commandLineMap).find((mangled) =>
+                arg.startsWith(mangled),
```

Comment:
I was going to ask about the use of .startsWith here, but I can see below that the args are expected to be in the form `["--arg1Name=arg1Value", ...]`

How consistent is this? Is it possible to get args instead as `["--arg1Name", "arg1Value", ...]`?

Reply:
(it might be more correct to use ``arg.startsWith(`${mangled}=`),`` instead)